src/app/codegen-data-model-provider/CodegenDataModelProvider_Write.cpp

```diff
@@ -322,7 +319,43 @@ DataModel::ActionReturnStatus CodegenDataModelProvider::WriteAttribute(const Dat
     if (!request.operationFlags.Has(DataModel::OperationFlags::kInternal))
     {
         VerifyOrReturnError(!isReadOnly, Status::UnsupportedWrite);
+    }
+
+    // ACL check for non-internal requests
+    bool checkAcl = !request.operationFlags.Has(DataModel::OperationFlags::kInternal);
 
+    // For chunking, ACL check is not re-done if the previous write was successful for the exact same
+    // path. We apply this everywhere as a shortcut, although realistically this is only for AccessControl cluster
+    if (checkAcl && request.previousSuccessPath.has_value())
+    {
+        checkAcl = ((request.path.mEndpointId != request.previousSuccessPath->mEndpointId) ||
+                    (request.path.mClusterId != request.previousSuccessPath->mClusterId) ||
+                    (request.path.mAttributeId != request.previousSuccessPath->mAttributeId));
```

Comment:
Why can't you just `*request.previousSuccessPath != request.path`?  I think you can, but if you can't it should be documented why so someone doesn't come along and "simplify" the code.

Reply:
Updated to do simple compare, however I find it unintuitive:
  - request.path is a data path, so it has a mListOp and mListIndex (which will NOT be part of the compare)
  - all ConcreteAttributePath have a `mExpanded` member which is also not part of compares (which also seems ugly to me: class member that is mutable but does not participate in equality/inequality.
  
 Added a comment explaining.

Reply:
It turns out they are different. What I belive is going on:

- `operator==` is defined on ConcreteDataAttributePath
- ConcreteDataAttributePath has a constructor from ConcreteAttributePath (which is lastSuccessPath)
- So we compare a data path (with list index and operation) with an auto-cast (which will zero out operation and index) so compares on acl writes on lists for example fails

Will add a comment and keep the compare code.

Reply:
Updated, hopefully this is acceptable.